examples/common/pigweed/rpc_services/Device.h

```diff
@@ -35,6 +36,175 @@
+class CommissionableDataProviderRpcWrapper : public DeviceLayer::CommissionableDataProvider
+{
+public:
+    CHIP_ERROR GetSetupPasscode(uint32_t & setupPasscode) override
+    {
+        LazyInit();
+        if (mPasscodeOverride.has_value())
+        {
+            setupPasscode = mPasscodeOverride.value();
+            return CHIP_NO_ERROR;
+        }
+        else if (mCommissionableDataProvider)
+        {
+            return mCommissionableDataProvider->GetSetupPasscode(setupPasscode);
+        }
+        return CHIP_ERROR_INTERNAL;
+    }
+
+    // NOTE: Changing the passcode will not change the verifier or anything else
+    // this just changes the value returned from GetSetupPasscode.
+    // Using this is completely optional, and only really useful for test
+    // automation which can read the configured passcode for commissioning
+    // after it is changed.
+    CHIP_ERROR SetSetupPasscode(uint32_t setupPasscode) override
+    {
+        LazyInit();
+        mPasscodeOverride = setupPasscode;
+        return CHIP_NO_ERROR;
+    }
+
+    CHIP_ERROR GetSetupDiscriminator(uint16_t & setupDiscriminator) override
+    {
+        LazyInit();
+        if (mDiscriminatorOverride.has_value())
+        {
+            setupDiscriminator = mDiscriminatorOverride.value();
+            return CHIP_NO_ERROR;
+        }
+        else if (mCommissionableDataProvider)
+        {
+            return mCommissionableDataProvider->GetSetupDiscriminator(setupDiscriminator);
+        }
+        return CHIP_ERROR_INTERNAL;
+    }
+
+    CHIP_ERROR SetSetupDiscriminator(uint16_t setupDiscriminator) override
+    {
+        LazyInit();
+        mDiscriminatorOverride = setupDiscriminator;
+        return CHIP_NO_ERROR;
+    }
+
+    CHIP_ERROR GetSpake2pIterationCount(uint32_t & iterationCount) override
+    {
+        LazyInit();
+        if (mIterationCountOverride.has_value())
+        {
+            iterationCount = mIterationCountOverride.value();
+            return CHIP_NO_ERROR;
+        }
+        else if (mCommissionableDataProvider)
+        {
+            return mCommissionableDataProvider->GetSpake2pIterationCount(iterationCount);
+        }
+        return CHIP_ERROR_INTERNAL;
+    }
+
+    CHIP_ERROR SetSpake2pIterationCount(uint32_t iterationCount)
+    {
+        LazyInit();
+        mIterationCountOverride = iterationCount;
+        return CHIP_NO_ERROR;
+    }
+
+    CHIP_ERROR GetSpake2pSalt(MutableByteSpan & saltBuf) override
+    {
+        LazyInit();
+        if (mSaltOverride.has_value())
+        {
+            if (mSaltOverride.value().size() > saltBuf.size())
+            {
+                return CHIP_ERROR_BUFFER_TOO_SMALL;
+            }
+            std::copy(mSaltOverride.value().begin(), mSaltOverride.value().end(), saltBuf.begin());
+            saltBuf.reduce_size(mSaltOverride.value().size());
+            return CHIP_NO_ERROR;
+        }
+        else if (mCommissionableDataProvider)
```

Comment:
nit: consider not having an else after return (if above if block is true, it will always return, else here is not necessary)

Reply:
done